File: packages/agent-sdk-ts/examples/request-credential.ts
// Request a credential from Emberlink.
// Run: npx tsx examples/request-credential.ts
// Requires: emberlink-agent binary in PATH and a provisioned database.

import { EmberAgent, EmberAgentError } from '../src/index.js';

const agent = new EmberAgent({
  dbPath: process.env.EMBERLINK_DB ?? './emberlink.db',
  personaId: process.env.EMBERLINK_PERSONA ?? 'my-agent-persona',
  daemonSocket: process.env.EMBERLINK_DAEMON_SOCKET,
});

try {
  await agent.connect();
} catch (err) {
  console.error('Could not start emberlink-agent. Is the binary in your PATH?', err);
  process.exit(1);
}

const me = await agent.whoami();
console.log('Connected as:', me.persona_id);

const grant = await agent.requestGrant({
  scope: 'repo:read',
  resource_id: process.env.EMBERLINK_CREDENTIAL ?? 'github-token',
  reason: 'streaming session',
});
console.log('Grant request submitted:', grant.request_id, '— status:', grant.status);

const status = await agent.grantStatus(grant.request_id);
console.log('Grant status:', status.status);

agent.close();

File: packages/agent-sdk-py/examples/request_credential.py
# Request a credential from Emberlink.
# Run: python examples/request_credential.py
# Requires: emberlink-agent binary in PATH and a provisioned database.

import os
import sys
from emberlink_agent import AgentConfig, EmberAgent, EmberAgentError

config = AgentConfig(
    db_path=os.environ.get("EMBERLINK_DB", "./emberlink.db"),
    persona_id=os.environ.get("EMBERLINK_PERSONA", "my-agent-persona"),
    daemon_socket=os.environ.get("EMBERLINK_DAEMON_SOCKET"),
)

try:
    with EmberAgent(config) as agent:
        me = agent.whoami()
        print("Connected as:", me.persona_id)

        grant = agent.request_grant(
            scope="repo:read",
            resource_id=os.environ.get("EMBERLINK_CREDENTIAL", "github-token"),
            reason="streaming session",
        )
        print("Grant request submitted:", grant.request_id, "— status:", grant.status)

        status = agent.grant_status(grant.request_id)
        print("Grant status:", status.status)

except FileNotFoundError:
    print("Could not start emberlink-agent. Is the binary in your PATH?", file=sys.stderr)
    sys.exit(1)
except EmberAgentError as e:
    print(f"Agent error [{e.code}]: {e.message}", file=sys.stderr)
    sys.exit(1)
